mautic/mautic
# Removes redundant code by adding a twig macro to `FormBundle`

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR adds a twig macro to Mautics templating engine (TWIG). With this macro the code for the HTML form creation becomes more readable as this allows to remove duplicate code. This PR does not change anything you see on the page/form, though.

**Changes**
- New Macro: Creates a macro called tag_attr in `app/bundles/FormBundle/Resources/views/Field/macros.html.twig`.
- Template Updates and Code Clean-up: Modifies existing `Field` twig templates to use this new macro. This change removed redundant code blocks previously used for attribute generation, reducing repetitive code and enhancing twig template readability.

#### Steps to test this PR:

1. Create a form with (ideally) all available input fields.
2. Open this form in a browser window.
3. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
4. Edit the form slightly. I.e. update a field label. This makes mautic recreated the cached HTML.
5. Open the form in  **a second** browser window.
6. Compare the looks. You should **not** see any change.
7. If you have time: Open the browsers developer console or the pages HTML source view: Compare the created HTML. There should not be any changes.
8. Enter values in every form input and submit the form. There must not be an error and the form submission should include all values.

Please review and test this PR. I created several form fields and they still work as expected, but I may have missed a field.
